### PR TITLE
Fix links in demo features and integrations pages

### LINKED ISSUE #2361: Jaeger link is invalid in integrations page
_Originally posted by @chalin in https://github.com/open-telemetry/opentelemetry.io/pull/2201#discussion_r1111078211_

---

@svrnm, the Jaeger link is no longer valid:

```console
$ curl -I https://www.jaegertracing.io/docs/latest/opentelemetry/
HTTP/2 301 
...
location: /docs/1.42/opentelemetry/
...
$ curl -I https://www.jaegertracing.io/docs/1.42/opentelemetry/
HTTP/2 404 
```

From https://www.jaegertracing.io/docs/1.42/getting-started/:

> Historically, the Jaeger project supported its own SDKs (aka tracers, client libraries) that implemented the OpenTracing API. The [Client Libraries](https://www.jaegertracing.io/docs/1.42/client-libraries) section provides information about how to use the OpenTracing API and how to initialize and configure Jaeger tracers. As of 2022, the Jaeger SDKs are no longer supported, and all users are advised to migrate to OpenTelemetry.

### PR DESCRIPTION
**Pull request by chalin**
- Closes #2361
- Contributes to #2429

/cc @svrnm

### CODE REVIEW

File: content/en/docs/demo/features.md [hunk 1]
```diff
@@ -4,25 +4,24 @@ linkTitle: Features
...
   Kubernetes (both locally, as well as on the cloud) using a Helm chart.
 - **[Docker](https://docs.docker.com)**: this forked sample can also be executed
   only with Docker.
-- **[gRPC](https://grpc.io)**: microservices use a high volume of gRPC calls to
+- **[gRPC](https://grpc.io/)**: microservices use a high volume of gRPC calls to
   communicate to each other.
 - **[HTTP](https://www.rfc-editor.org/rfc/rfc9110.html)**: microservices use
   HTTP where gRPC is unavailable or not well supported.
-- **[OpenTelemetry Traces](https://opentelemetry.io)**: all services are
+- **[OpenTelemetry Traces](/docs/concepts/signals/traces/)**: all services are
```
Comment (chalin) [inline content/en/docs/demo/features.md:15]: Note the change in URL from the site homepage to the Traces page under Concepts.

File: content/en/docs/demo/features.md [hunk 2]
```diff
@@ -4,25 +4,24 @@ linkTitle: Features
...
   instrumented using OpenTelemetry available instrumentation libraries.
-- **[OpenTelemetry Metrics](https://opentelemetry.io)**: Select services are
-  instrumented using OpenTelemetry available instrumentation libraries. More
+- **[OpenTelemetry Metrics](/docs/concepts/signals/metrics/)**: Select services
```
Comment (chalin) [inline content/en/docs/demo/features.md:17]: Note the change in URL from the site homepage to the Metrics page under Concepts.

File: content/en/docs/demo/features.md [hunk 3]
```diff
@@ -4,25 +4,24 @@ linkTitle: Features
...
+  are instrumented using OpenTelemetry available instrumentation libraries. More
   will be added as the relevant SDKs are released.
-- **[OpenTelemetry Collector](/docs/collector/getting-started)**: all services
-  are instrumented and sending the generated traces and metrics to the
-  OpenTelemetry Collector via gRPC. The received traces are then exported to the
-  logs and to Jaeger; received metrics and exemplars\* are exported to logs and
-  Prometheus.
-- **[Jaeger](https://www.jaegertracing.io)**: all generated traces are being
+- **[OpenTelemetry Collector](/docs/collector/)**: all services are instrumented
```
Comment (chalin) [inline content/en/docs/demo/features.md:20]: Note the change in path from the Collector Getting-started page to the Collector landing page.

File: content/en/ecosystem/integrations.md [hunk 4]
```diff
@@ -38,7 +38,7 @@ The following open source projects use OpenTelemetry components.
 | [containerd](https://github.com/containerd/containerd/blob/main/docs/tracing.md)           | Go                                  |
 | [CRI-O](https://github.com/cri-o/cri-o/blob/main/docs/crio.conf.5.md#criotracing-table)    | Go                                  |
 | [Docker buildx](https://github.com/docker/buildx/blob/master/docs/guides/opentelemetry.md) | Go                                  |
-| [Jaeger](https://www.jaegertracing.io/docs/latest/opentelemetry/)                          | Collector                           |
+| [Jaeger](https://www.jaegertracing.io/)                                                    | Collector, Go                       |
```
Comment (chalin) [inline content/en/ecosystem/integrations.md:41]: Fix for #2361

**Review by svrnm — APPROVED** [review]
lgtm

**Comment by chalin** [comment]
@yurishkuro - feedback is still welcome (I'll address it in a followup PR, as necessary).

**Pull request merged**